vendor/github.com/hashicorp/hcl/v2/hclsyntax/parser_traversal.go

```diff
@@ -64,11 +64,56 @@ func (p *parser) ParseTraversalAbs() (hcl.Traversal, hcl.Diagnostics) {
 				return ret, diags
 			}
 
-			attrName := string(nameTok.Bytes)
-			ret = append(ret, hcl.TraverseAttr{
-				Name:     attrName,
-				SrcRange: hcl.RangeBetween(dot.Range, nameTok.Range),
-			})
+			if nameTok.Type == TokenIdent {
```

Comment:
example of upstream changes - I wanted to get feedback on this POC before opening a PR there.